### PR TITLE
WidgetGallery: Convert layout to GML and add new Icons gallery

### PR DESCRIPTION
**Pull request by thankyouverycool**
https://user-images.githubusercontent.com/66646555/110861337-feb0d980-828b-11eb-9fdf-0cc0d7ce870a.mp4

### CODE REVIEW

File: Userland/Demos/WidgetGallery/GalleryGML/BasicsTab.gml [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+                }
+            }
+        }
+
+        @GUI::HorizontalSeparator {
+        }
+
+        @GUI::Widget {
+            fixed_height: 22
+            layout: @GUI::HorizontalBoxLayout{
```
Comment (emanuele6) [inline Userland/Demos/WidgetGallery/GalleryGML/BasicsTab.gml:57]:
```suggestion
            layout: @GUI::HorizontalBoxLayout {
```

File: Userland/Demos/WidgetGallery/GalleryGML/BasicsTab.gml [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+            @GUI::VerticalSeparator {
+            }
+
+            @GUI::Widget {
+                layout: @GUI::VerticalBoxLayout {
+                }
+
+                @GUI::TextBox {
+                        text: "Read only"
+                        mode: "ReadOnly"
```
Comment (emanuele6) [inline Userland/Demos/WidgetGallery/GalleryGML/BasicsTab.gml:272]:
```suggestion
                    text: "Read only"
                    mode: "ReadOnly"
```

**Review by emanuele6 — CHANGES_REQUESTED** [review]

File: Userland/Demos/WidgetGallery/GalleryGML/BasicsTab.gml [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+            @GUI::Widget {
+                layout: @GUI::VerticalBoxLayout {
+                    margins: [4, 0, 4, 0]
+                }
+
+                @GUI::Label {
+                    name: "word_wrap_label"
+                    word_wrap: true
+                    text_alignment: "TopLeft"
+                    text: "Lorem ipsum sistema serenitas, per construxit klingre sed quis awesomatia, ergo salve amici."
```
Comment (emanuele6) [inline Userland/Demos/WidgetGallery/GalleryGML/BasicsTab.gml:47]: :rofl:

File: Userland/Demos/WidgetGallery/GalleryModels.h [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+
+#include <AK/NonnullRefPtr.h>
+#include <AK/Vector.h>
+#include <LibCore/DirIterator.h>
+#include <LibGUI/Model.h>
+#include <LibGUI/ModelRole.h>
+#include <LibGfx/Bitmap.h>
+
+template<typename T>
+class ListViewModel final : public GUI::Model {
```
Comment (alimpfard) [inline Userland/Demos/WidgetGallery/GalleryModels.h:37]: `GUI::ItemListModel<T>`?

Comment (thankyouverycool) [reply Userland/Demos/WidgetGallery/GalleryModels.h]: ahh yes thanks! I knew I should have looked harder xD

**Comment by tomuta** [comment]
Thank you, very cool. Indeed :)

**Pull request merged**